source/extensions/omni.isaac.orbit/omni/isaac/orbit/devices/gamepad/se3_gamepad.py

```diff
@@ -0,0 +1,203 @@
+        # command buffers
+        self._close_gripper = False
+        self._delta_pos = np.zeros(3)  # (x, y, z)
+        self._delta_rot = np.zeros(3)  # (roll, pitch, yaw)
+        # dictionary for additional callbacks
+        self._additional_callbacks = dict()
+
+    def __str__(self) -> str:
+        """Returns: A string containing the information of joystick."""
+        msg = f"Gamepad Controller for SE(3): {self.__class__.__name__}\n"
+        msg += "\tToggle gripper (open/close): B/A\n"
+        msg += "\tMove arm along x-axis: Left Stick Up/Down\n"
+        msg += "\tMove arm along y-axis: Left Stick Left/Right\n"
+        msg += "\tMove arm along z-axis: Right Stick Up/Down\n"
+        msg += "\tRotate arm along x-axis: D-Pad Left/Right\n"
+        msg += "\tRotate arm along y-axis: D-Pad Up/Down\n"
+        msg += "\tRotate arm along z-axis: Right Stick Left/Right\n"
+        return msg
+
+    """
+    Operations
+    """
+
+    def reset(self):
+        # default flags
+        self._close_gripper = False
+        self._delta_pos = np.zeros([2, 3])  # (pos, neg) (x, y, z)
+        self._delta_rot = np.zeros([2, 3])  # (pos, neg) (roll, pitch, yaw)
+
+    def add_callback(self, key: str, func: Callable):
+        """Add additional functions to bind gamepad.
+
+        A list of available keys are present in the
+        `carb documentation <https://docs.omniverse.nvidia.com/kit/docs/carbonite/latest/docs/python/carb.html?highlight=gamepadeventtype#carb.input.GamepadInput>`_.
+
+        The callback function should not take any arguments.
+
+        Args:
+            key (str): The gamepad button to check against.
+            func (Callable): The function to call when key is pressed.
+        """
+        self._additional_callbacks[key] = func
+
+    def advance(self) -> Tuple[np.ndarray, bool]:
+        """Provides the result from gamepad event state.
+
+        Returns:
+            Tuple[np.ndarray, bool] -- A tuple containing the delta pose command and gripper commands.
+        """
+        delta_rot_sgn = self._delta_rot[0, :] > self._delta_rot[1, :]
+        delta_rot = self._delta_rot.max(axis=0)
+        delta_rot[~delta_rot_sgn] *= -1
+        delta_pos_sgn = self._delta_pos[0, :] > self._delta_pos[1, :]
+        delta_pos = self._delta_pos.max(axis=0)
+        delta_pos[~delta_pos_sgn] *= -1
+
+        rot_vec = Rotation.from_euler("XYZ", delta_rot).as_rotvec()
+        # if new command received, reset event flag to False until gamepad updated.
+        return np.concatenate([delta_pos, rot_vec]), self._close_gripper
+
+    """
+    Internal helpers.
+    """
+
+    def _on_gamepad_event(self, event, *args, **kwargs):
+        """Subscriber callback to when kit is updated.
+
+        Reference:
+            https://docs.omniverse.nvidia.com/kit/docs/carbonite/latest/docs/python/carb.html?highlight=gamepadeventtype#carb.input.Gamepad
+        """
+
+        cur_val = event.value
+        absval = abs(event.value)
```

Comment:
Also here. Please add docstring to explain why we need to take ab abs value

Reply:
Thanks for your question. I thought about it again and found that it is not necessary for this `absval` so I removed it in the new commit